### PR TITLE
#3936 rework universe tests

### PR DESCRIPTION
**Pull request by bthaile**
more mocks! older tests didn't leverage any business logic of other contracts, so I reworked the tests to use mocks.  added call verification on augur contract using mock augur.

### CODE REVIEW

File: tests/reporting/test_universe.py [hunk 1]
```diff
@@ -3,162 +3,431 @@
...
+    repPriceOracle = localFixture.uploadAndAddToController("../source/contracts/reporting/RepPriceOracle.sol", 'repPriceOracle')
+    controller.setValue(stringToBytes('RepPriceOracle'), repPriceOracle.address)
+    
+    multiplier = localFixture.contracts['Constants'].TARGET_REP_MARKET_CAP_MULTIPLIER()
+    # default value
+    defaultValue = 10000
+    assert populatedUniverse.getRepMarketCapInAttoeth() == 0
+    assert populatedUniverse.getReportingFeeDivisor() == defaultValue
+    
+    # puse reporting window forward
```
Comment (nuevoalex) [inline tests/reporting/test_universe.py]: s/puse/push

Comment (nuevoalex) [reply tests/reporting/test_universe.py]: elsewhere too ;p

Comment (bthaile) [reply tests/reporting/test_universe.py]: thanks got it.

File: tests/solidity_test_helpers/MockAugur.sol [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    event MarketFinalized(address indexed universe, address indexed market);
+    event UniverseForked(address indexed universe);
+    event OrderCanceled(address indexed universe, address indexed shareToken, address indexed sender, bytes32 orderId, Order.TradeTypes orderType, uint256 tokenRefund, uint256 sharesRefund);
+    event OrderCreated(address indexed universe, address indexed shareToken, address indexed creator, bytes32 orderId, uint256 tradeGroupId);
+    event OrderFilled(address indexed universe, address indexed shareToken, address filler, bytes32 orderId, uint256 numCreatorShares, uint256 numCreatorTokens, uint256 numFillerShares, uint256 numFillerTokens, uint256 settlementFees, uint256 tradeGroupId);
+    event ProceedsClaimed(address indexed universe, address indexed shareToken, address indexed sender, address market, uint256 numShares, uint256 numPayoutTokens, uint256 finalTokenBalance);
+    event UniverseCreated(address indexed parentUniverse, address indexed childUniverse);
+    event TokensTransferred(address indexed universe, address indexed token, address indexed from, address to, uint256 value);
```
Comment (nuevoalex) [inline tests/solidity_test_helpers/MockAugur.sol]: Seems like we can remove these if they're never used

Comment (bthaile) [reply tests/solidity_test_helpers/MockAugur.sol]: I don't know if they will be used. I'll remove them and add them back if need be

**Pull request merged**